src/plugins/intel_gpu/src/runtime/ocl/ocl_engine.cpp

```diff
@@ -127,30 +127,35 @@ allocation_type ocl_engine::detect_usm_allocation_type(const void* memory) const
 
 bool ocl_engine::check_allocatable(const layout& layout, allocation_type type) {
     OPENVINO_ASSERT(supports_allocation(type) || type == allocation_type::cl_mem, "[GPU] Unsupported allocation type: ", type);
-    auto alloc_mem_size = layout.bytes_count();
-    auto max_mem_size = get_device_info().max_alloc_mem_size;
-    if (alloc_mem_size > max_mem_size) {
-        auto used_mem = get_used_device_memory(allocation_type::usm_device) + get_used_device_memory(allocation_type::usm_host);
-        GPU_DEBUG_LOG << "[GPU] Mem size info: " << "Required " << alloc_mem_size << " bytes, already occupied : "
-                      << used_mem << " bytes, available memory size is " << get_max_memory_size() << " bytes, but max allocable memory size is "
-                      << max_mem_size << " bytes." << std::endl;
+
+    OPENVINO_ASSERT(layout.bytes_count() <= get_device_info().max_alloc_mem_size,
+                    "[GPU] Exceeded max size of memory object allocation: ",
+                    "Requested ", layout.bytes_count(), " bytes "
+                    "but max alloc size is ", get_device_info().max_alloc_mem_size, " bytes");
```

Comment:
How about more user-friendly messages with some suggestions? Something like: 
`"[GPU] Exceeded max size of memory object allocation: requested N bytes, but max alloc size supported by device is M bytes. Please try to reduce batch size or use lower precision"`